rfcs/convergence/input-native-element-props.md

```diff
@@ -0,0 +1,192 @@
+# RFC: How to handle native element props for input components
+
+<!--
+An RFC can be anything. A question, a suggestion, a plan. The purpose of this template is to give some structure to help folks write successful RFCs. However, don't feel constrained by this template; use your best judgement.
+
+Tips for writing a successful RFC:
+
+- Simple plain words that make your point, fancy words obfuscate
+- Try to stay concise, but don't gloss over important details
+- Try to write a neutral problem statement, not one that motivates your desired solution
+- Remember, "Writing is thinking". It's natural to realize new ideas while writing your proposal
+-->
+
+---
+
+@sopranopillow @ecraig12345
+
+## Summary
+
+This RFC proposes options on how to handle the native element props for input components on `vNext`. Currently the native props for all vNext components are applied to the root element, but in the case of an input component there are native input props that should be handled as well.
+
+<!-- Explain the proposed change -->
+
+## Problem statement
+
+For most components, it makes sense to apply native props (and the ref) to the root element.
+
+The input components are a little unique because the actual `<input>` *cannot* be the root element of the component, since we need various wrappers for styling. For example, rendering a `<Checkbox/>` actually does this:
+```tsx
+<div>
+  <div ... /> {/* checkmark */}
+  <input type="checkbox" />
+  <label {...slotProps.label} />
+</div>
+```
+Or an `<Input/>` is roughly like this:
+```tsx
+<div>
+  {/* optional bookend before slot */}
+  <div> {/* wrapper visually styled as an input (maybe, TBD) */}
+    {/* optional start slot */}
+    <input type="text" .../>
+    {/* optional end slot */}
+  </div>
+  {/* optional bookend after slot */}
+</div>
+```
+
+Another unique thing about inputs is that to the degree possible, we'd like them to work nicely with 3rd-party form libraries, which may have APIs that expect to be able to pass native props to the component and have them applied to the actual `<input>`. (needs more research)
```

Comment:
regarding proper support of various libraries/frameworks - it might be good idea to publish fluent wrappers as separate packages for those particular solutions to make adoption super easy